mapbox/libandroid-telemetry/src/main/java/com/mapbox/services/android/telemetry/navigation/MapboxNavigationEvent.java

```diff
@@ -0,0 +1,169 @@
+package com.mapbox.services.android.telemetry.navigation;
+
+import android.location.Location;
+
+import com.mapbox.services.android.telemetry.constants.TelemetryConstants;
+import com.mapbox.services.android.telemetry.utils.TelemetryUtils;
+
+import java.util.Date;
+import java.util.Hashtable;
+
+/**
+ * Represent events dispatched by the Navigation SDK.
+ */
+public class MapboxNavigationEvent {
+
+  private static final int EVENT_VERSION = 1;
+
+  // Event types
+  public static final String TYPE_TURNSTILE = "navigation.turnstile";
+  public static final String TYPE_DEPART = "navigation.depart";
+  public static final String TYPE_FEEDBACK = "navigation.feedback";
+  public static final String TYPE_ARRIVE = "navigation.arrive";
+  public static final String TYPE_CANCEL = "navigation.cancel";
+
+  // Event keys
+  public static final String KEY_EVENT = "event";
+  public static final String KEY_PLATFORM = "platform";
```

Comment:
We've removed platform from the spec since it can be derived from the operating system/SDK identifier.